### PR TITLE
WIP initial log structure

### PR DESCRIPTION
**Pull request by JGcarv**
Just a WIP PR to see if I'm in the correct line of implementation, before I spend quite some time replacing the printf statements!

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -265,17 +282,17 @@ func mineCmd(cmd *cli.Cmd) {
 		DB := ctx.Value(tellorCommon.DataProxyKey).(db.DataServerProxy)
 		v, err := DB.Get(db.DisputeStatusKey)
 		if err != nil {
-			fmt.Println("ignoring --- could not get dispute status.  Check if staked")
+			level.Warn(logger).Log("msg", "ignoring --- could not get dispute status.  Check if staked")
 		}
 		status, _ := hexutil.DecodeBig(string(v))
 		if status.Cmp(big.NewInt(1)) != 0 {
-			log.Fatalf("Miner is not able to mine with status %v. Stopping all mining immediately", status)
+			level.Error(logger).Log("err", "Miner is not able to mine with status %v. Stopping all mining immediately", status)
```
Comment (krasi-georgiev) [inline main.go:289]: lets be consistent and use `msg` and OTOH in the logger we need to have an even number of argument or use `fmt.Sprintf`

so here you can either have 
`level.Error(logger).Log("msg", fmt.Sprintf("Miner is not able to mine with status %v. Stopping all mining immediately", status))`

or 

`level.Error(logger).Log("msg", "Miner is not able to mine. Stopping all mining immediately","status", status))`
I am a fan of this second version but don't mind either way.

also when you see `log.Fatalf` this means that the should also exit after logging so here you should also add `os.Exit(1)` after the log line.

File: ops/disputeOps.go [hunk 2]
```diff
@@ -205,6 +207,7 @@ func List(ctx context.Context) error {
 
 	instance := ctx.Value(tellorCommon.MasterContractContextKey).(*tellor.TellorMaster)
 
+	// Not sure if we should handle this with logger??
```
Comment (krasi-georgiev) [inline ops/disputeOps.go:210]: no this one is fine to leave like this as the comment is to just list disputes

File: ops/miningManager.go [hunk 3]
```diff
@@ -40,11 +42,12 @@ type MiningMgr struct {
 }
 
 // CreateMiningManager creates a new manager that mananges mining and data requests.
-func CreateMiningManager(ctx context.Context, exitCh chan os.Signal, submitter tellorCommon.TransactionSubmitter) (*MiningMgr, error) {
+func CreateMiningManager(ctx context.Context, exitCh chan os.Signal, submitter tellorCommon.TransactionSubmitter, logger log.Logger) (*MiningMgr, error) {
```
Comment (krasi-georgiev) [inline ops/miningManager.go:45]: looks like you missed my chat comments :smile_cat: 
In golang as standard first argument is context and second is logger.

```suggestion
func CreateMiningManager(ctx context.Context, logger log.Logger, exitCh chan os.Signal, submitter tellorCommon.TransactionSubmitter) (*MiningMgr, error) {
```

File: ops/miningManager.go [hunk 4]
```diff
@@ -40,11 +42,12 @@ type MiningMgr struct {
...
 	cfg := config.GetConfig()
 
 	group, err := pow.SetupMiningGroup(cfg)
 	if err != nil {
+
```
Comment (krasi-georgiev) [inline ops/miningManager.go:50]:
```suggestion
```

**Review by krasi-georgiev — COMMENTED** [review]
Thanks @JGcarv very good start! 
I added some initial comments.

File: main.go [hunk 5]
```diff
@@ -146,11 +156,14 @@ func App() *cli.Cli {
 	configPath := app.StringOpt("config", "configs/config.json", "Path to the primary JSON config file")
 	logPath := app.StringOpt("logConfig", "configs/loggingConfig.json", "Path to a JSON logging config file")
 
+	//Ignoring loging Config for now
```
[the remark this answers is not in the document]
Comment (JGcarv) [reply main.go:159]: The reason I'm ignoring for now is that the log config allows to set different log filters for each package, which I couldn't quite figure it out how to do yet.

File: ops/miningManager.go (hunk 3 above)
Comment (JGcarv) [reply ops/miningManager.go:45]: Ops, I actually missed that. I'll adjust it

File: main.go [hunk 6]
```diff
@@ -213,27 +226,31 @@ func disputeCmd(cmd *cli.Cmd) {
 }
 
 func voteCmd(cmd *cli.Cmd) {
+	logger := setupLogger()
```
[the remark this answers is not in the document]
Comment (JGcarv) [reply main.go:229]: The this is that for some reason I can't pass any argument to the functions that end with `Cmd` and take the `cmd *cli.Cmd` as a first argument. 

I initially set it up to config on here and pass to all the functions below:
```	app.Command("stake", "staking operations", stakeCmd)
	app.Command("transfer", "send TRB to address", moveCmd(ops.Transfer))
	app.Command("approve", "approve TRB to address", moveCmd(ops.Approve))
	app.Command("balance", "check balance of address", balanceCmd)
	app.Command("dispute", "dispute operations", disputeCmd)
	app.Command("mine", "mine for TRB", mineCmd)
	app.Command("dataserver", "start an independent dataserver", dataserverCmd)
```

But I can't figure it out how to do that

File: main.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (krasi-georgiev) [reply main.go:159]: tbh I don't think we even need separate configs for each component, lets just use a single log config and can be even passes as cli args like in Thanos.

File: main.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (krasi-georgiev) [reply main.go:229]: you can do it like this.

```
func disputeCmd(logger log.Logger, cmd *cli.Cmd) {
	cmd.Command("vote", "vote on an active dispute", voteCmd(logger))
	...
	..
}

func voteCmd(logger) *cli.Cmd {
	return func(cmd *cli.Cmd) {
		disputeID := EthereumInt{}
		cmd.VarArg("DISPUTE_ID", &disputeID, "dispute id")
		supports := cmd.BoolArg("SUPPORT", false, "do you support the dispute? (true|false)")
		cmd.Action = func() {
			ErrorHandler(ops.Vote(disputeID.Int, *supports, ctx, logger), "vote")
		}
	}
}
```

it is ugly but lets keep it with minimum refactoring and will fix this in follow up prs.

**Comment by JGcarv** [comment]
I Believe I addresses most of this issues in the other PR. This is outdated by now!

**Pull request closed without merging**